src/averageAges.js

```diff
@@ -54,8 +75,22 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = onlyWithSon
+  // eslint-disable-next-line max-len
+    ? people.filter(person => people.some(mother => mother.name === person.mother) && person.sex === 'm')
+  // eslint-disable-next-line max-len
+    : people.filter(person => people.some(mother => mother.name === person.mother));
```

Comment:
Remove all disabled & create a separate fucntion for this
```suggestion
    ? people.filter(person => hasChild(people, person.mother) && person.sex === 'm')
    : people.filter(person => hasChild(people, person.mother));
```